interpreter.js

```diff
@@ -2807,8 +2807,24 @@ Interpreter.prototype['stepCallExpression'] = function() {
   if (!state.doneExec_) {
     state.doneExec_ = true;
     var func = state.func_;
-    if (!func || !func.isObject) {
-      this.throwException(this.TYPE_ERROR, func + ' is not a function');
+    var isNotFunction = ' is not a function';
+    if (!func) {
+      var callee = node['callee'];
+      var components = state.components_;
+      var functionName;
+      if (callee.object) {
+        functionName = callee.object.name;
+        if (components && Array.isArray(components) && 1 <= components.length) {
+          functionName += '.' + components[1];
```

Comment:
it's still not clear to me that this is the correct way to deal with this. For example, with 

```js
var foo = {};
foo.bar = {};
foo.bar.baz();
```

We now get `TypeError: undefined.baz is not a function`

What specifically is our goal with this change?

Reply:
Thanks for pairing on this. Let me know what you think of the refactor. It handles cases of nested objects as well as function calls on variables.